src/main/java/io/openepcis/convert/VersionTransformer.java

```
@@ -0,0 +1,257 @@
+          "application/xml",
+          EpcisVersion.VERSION_2_0,
+          EpcisVersion.VERSION_1_2);
+    } else if (fromMediaType.toLowerCase().contains("xml")
+        && toMediaType.toLowerCase().contains("json")
+        && fromVersion.equals(EpcisVersion.VERSION_2_0)
+        && toVersion.equals(EpcisVersion.VERSION_2_0)) {
+      // If fromMedia is xml and toMedia is json and both versions are 2.0 convert xml->json
+      return toJson(inputDocument);
+    } else if (fromMediaType.toLowerCase().contains("xml")
+        && toMediaType.toLowerCase().contains("json")
+        && fromVersion.equals(EpcisVersion.VERSION_1_2)
+        && toVersion.equals(EpcisVersion.VERSION_2_0)) {
+      // If fromMedia is xml and toMedia is json and fromVersion is 1.2, toVersion 2.0 then convert
+      // xml->2.0 and then to JSON
+      final InputStream convertedXml =
+          convert(
+              inputDocument, "application/xml", EpcisVersion.VERSION_1_2, EpcisVersion.VERSION_2_0);
+      return toJson(convertedXml);
+    } else {
+      throw new UnsupportedOperationException(
+          "Requested conversion is not supported, Please check provided MediaType/Version and try again");
+    }
+  }
+
+  // Private method to convert the JSON 2.0 document -> XML 2.0 and return it as InputStream
+  private InputStream toXml(final InputStream inputDocument) {
+    try {
+      final PipedOutputStream xmlOutputStream = new PipedOutputStream();
+      final EventHandler<? extends XmlEpcisEventsCollector> handler =
+          new EventHandler(new XmlEpcisEventsCollector(xmlOutputStream));
+
+      final PipedInputStream convertedDocument = new PipedInputStream(xmlOutputStream);
+
+      Executors.newWorkStealingPool()
+          .execute(
+              () -> {
+                try {
+                  jsonToXmlConverter.convert(inputDocument, handler);
+                  xmlOutputStream.close();
+                } catch (Exception e) {
+                  e.printStackTrace();
```

don't "swallow" this Exception -> throw some useable exception